Add support for filtering by repository name

henrikra opened the pull request:
First of all great to see this great repo here!

I wanted to add feature that you can filter by repositories. You can filter by multiple repository names by using query param. Please read README also :)

henrikra committed:
Simplify query string parsing

henrikra committed:
Support filtering by repository name

henrikra committed:
Refactor to form so user can use enter key to submit the form

sampsakuronen commented on docs/index.html:
```diff
@@ -226,21 +236,15 @@ <h1 id="error"></h1>
       var serverUrlInput = document.getElementById('serverurl')
       var apiTokenInput = document.getElementById('apitoken')
-      var submit = document.getElementById('submit')
+      const form = document.querySelector('form');
```
We have kept away from `const` and all ES6 in order to provide all the possible running environments.

henrikra replied:
Why is that when support is over 95% *world wide* plus radiators are usually deployed just locally in controlled environment aka we can choose what ever browser :)

Here is the stats for `const`
https://caniuse.com/#search=const

sampsakuronen replied:
I think that we can look at this from a couple of different perspectives. This is the one I looked at it when writing the comments:

- All the current radiators are deployed "somewhere" with "some setup" that is not known to maintainers of this repository.
- Most of them are forks so they can opt-in to an update like this. We have also provided a "hosted" service as a homepage for this repository on Github Pages. The latter one is the problematic.
- We have had the non-ES6 level of browser support from the beginning
- Transition like this might not be backwards compatible for some
- Gains from transitioning to ES6 would benefit only the developers in this project – development alltogether is very sporadic and rare

I think that the (arguably miniscule) risk of breaking compatibility is more important than having an extra drop of elegance for the codebase. If an action to move to ES6 would be preferred it would have to happen for the whole codebase at the same time.

sampsakuronen commented on docs/index.html:
```diff
@@ -226,21 +236,15 @@ <h1 id="error"></h1>
-   }).fromPairs().value()
-   settings = extendWithDefaults(settings)
-
+   // main start
```
Remove comment

sampsakuronen commented on docs/backends.js:
```diff
@@ -23,12 +23,14 @@ function buildBackend(settings, callback) {
+            if (err) {
+                  return callback(err)
+            }
+            let builds = data.filter(repositoryFilter).filter(branchFilter);
```
Fix intendation

sampsakuronen commented on docs/backends.js:
```diff
@@ -23,12 +23,14 @@ function buildBackend(settings, callback) {
    var branchFilter = function(build) {
       return settings.branch ? build.branch.match(settings.branch) : true
    }
+   const repositoryFilter = build =>
```
We have stayed away from ES6. Use `var` and old school function syntax. This is to make sure that the radiator can run in pretty much any environment.

sampsakuronen commented on README.md:
```diff
@@ -36,6 +36,12 @@ any required parameters are missing.
+
+  Select which repositories you want to show in the radiator. Separated strings with commas
+  For example `my-cool-project,helloWorldProject`
+  Accepts all repositories if not specified
```
Improvement to text:

>Use for declaring wanted repositories in comma-separated format: `my-cool-project,helloWorldProject`
>By default all repositories will be shown

sampsakuronen reviewed (changes requested):
- Please see comments for things that need fixing. 
- Test thoroughly for 100 % backward compatibility.
- Smush commits so that they are more relevant for future developers as historical knowledge. Especially ones that reflect development time timeline/structure.

henrikra committed:
Fix few typos

henrikra committed:
Refactor ES5

henrikra commented:
All comments are now fixed! 🎉 
I squashed commits together so there are only few commits now. Also I refactored const to vars again. Although I disagree that this project couldn't use ES6 :D I would understand if this project would be used by thousands. Also I am pretty sure people would create issues if something broke :)

Ready for merge? Next thing I was going to do is to format whole project with Prettier something. Since it has mixed formatting now everywhere :D
